contracts/extensions/Whitelist.sol

```diff
@@ -0,0 +1,122 @@
+/*
+  This file is part of The Colony Network.
+
+  The Colony Network is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  The Colony Network is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with The Colony Network. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+pragma solidity 0.7.3;
+pragma experimental ABIEncoderV2;
+
+import "./ColonyExtension.sol";
+
+// ignore-file-swc-108
+
+
+contract Whitelist is ColonyExtension {
+
+  //  Events
+
+  event UserApproved(address _user, bool _status);
+  event AgreementSigned(address _user);
+
+  // Storage
+
+  bool useApprovals;
+  string agreement;
```

Comment:
Why change this to a string? It being a hash was good (fixed gas cost). If it's only to be able to tell when it's not initialised, I'd much rather have a separate boolean (which can go in the same slot as `useApprovals` to minimise gas).

Reply:
I was thinking about it and I realized that if we passed in a hash, we couldn't figure out from on-chain data where the original agreement was. By passing in a string, we can store the proper IPFS hash. Otherwise we'd need some other way for the frontend to figure it out.

Reply:
Okay, I definitely don't like the suggestion in the comments that this could be the full text, though.

Reply:
I mean, that's really up to the colony, all the gas costs are on them.

Reply:
Oh alright.